core/src/main/scala/kafka/utils/ZkUtils.scala

```diff
@@ -256,19 +251,41 @@ class ZkUtils(val zkClient: ZkClient,
   }
 
   /**
-   * Register brokers with v2 json format (which includes multiple endpoints).
+   * Register brokers with v3 json format (which includes multiple endpoints and rack).
```

Comment:
One thing I want to add is that we have to write v2 when protocol is 0.9.0.X because of KAFKA-3100. Having rack in v2 version is probably fine but I don't see any benefit of doing that.